backend/src/zimfarm_backend/routes/utils.py

```diff
@@ -23,24 +29,33 @@ def has_dict_sub_key(data: Dict[str, Any], keys: List[str]):
     return True
 
 
-def remove_secrets_from_response(response: dict):
+def remove_secrets_from_payload(payload: dict[str, Any]):
     """replaces or removes (in-place) all occurences of secrets"""
-    remove_secret_flags(response)
-    remove_url_secrets(response)
+    remove_secret_flags(payload)
+    remove_url_secrets(payload)
 
 
-def remove_secret_flags(response: dict):
+def remove_secret_flags(response: dict[str, Any]):
     """replaces (in-place) all occurences of secret flags with stars"""
 
     # we need flags to retrieve the secret value and replace it where we find it
     # in commands
-    if "config" not in response or "flags" not in response["config"]:
+    if "config" not in response or "offliner" not in response["config"]:
         return
 
-    schema = ScheduleConfigSchema.get_offliner_schema(
-        response["config"]["task_name"]
-    )().to_desc()
-    fields = [field["data_key"] for field in schema if field.get("secret", False)]
+    schema = ScheduleConfigSchema.get_offliner_schema(response["config"]["task_name"])
```

Comment:
Rather than calling this function, I feel like we should do it "Pydantic-way":
- use [Serialization Context](https://docs.pydantic.dev/latest/concepts/serialization/#serialization-context) to let serializers know when we want to hide secrets in model fields
- use custom serializers for `OptionalSecretStr`, `SecretStr` and `S3OptimizationCache` fields which will use this context to hide what needs to be hidden

Does it make any sense?